feature_engine/datetime/datetime.py

```diff
@@ -252,24 +261,44 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
 
         # check if dataset contains na
         if self.missing_values == "raise":
-            _check_contains_na(X, self.variables_)
+            if self.variables_ == ["index"]:
+                if X.index.isnull().any():
+                    raise ValueError(
+                        "Some of the variables to transform contain NaN. Check and "
+                        "remove those before using this transformer."
+                    )
 
-        # reorder variables to match train set
-        X = X[self.feature_names_in_]
+            else:
+                _check_contains_na(X, self.variables_)
 
-        # convert datetime variables
-        datetime_df = pd.concat(
-            [
+        if self.variables_ == ["index"]:
```

Comment:
to extract features from the index, do we absolutely need to make it a column?

 Can we not do df.index.month? or something like this? I saw it the other day on a stackoverflow thread.

But maybe it does not fit with how we structured the class?